### ISSUE TITLE
FR:  Add Support for Inline Footnotes for `footnote-after-punctuation` and `re-index-footnotes`

### ISSUE THREAD

**Issue by sampokantola** [platers/obsidian-linter#127]

**Comment by platers** [comment]
can you share an example? Would help me since I don't use footnotes myself :)

**Comment by sampokantola** [comment]
Inline footnotes are written as follows:
```
This is a sentence.^[This is an inline footnote!]
```

The difference to regular footnotes is that 1) the caret goes outside of the brackets, and 2) the footnote content is written inline inside the brackets instead of writing it in its own block elsewhere.

![image](https://user-images.githubusercontent.com/50581738/139514292-8d2899f6-d510-45c3-a928-f22b0f53c64a.png)

An example of inline footnotes is also included in the [Obsidian Help vault](https://help.obsidian.md/How+to/Format+your+notes#Footnotes).

**Comment by pjkaufman** [comment]
To clarify, @sampokantola , you are saying that the inline footnotes need to be left where they are in the file instead of having them moved to the end of the file, right?

**Comment by sampokantola** [comment]
@pjkaufman Yes, that's what I'm suggesting here: Move inline footnotes after punctuation, but still keep them inline.

(Also update re-index-footnotes to account for inline footnotes, so the regular ones after an inline footnote are incremented by 1 even though inline footnotes don't have a number in the markup.)

But I also had the idea for a new lint rule that would automatically convert inline footnotes into regular ones if they surpass a configurable character limit—but looks like I never made a feature request about that.
